homeworks/ReactProject/src/pages/loginPage/components/LoginForm.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import { Button, Form, Alert } from 'reactstrap';
+
+import { FormElementGroup } from '../../../components/FormElementGroup';
+
+export class LoginForm extends React.Component { 
+
+  state = {
+    data: {
+      login: '',
+      password: ''
+    },
+    errors: {}
+  }
+
+  handleChange = ({ target }) => {
+    this.setState({ data: { ...this.state.data, [target.name]  : target.value }});
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const errors = this.validate(this.state.data);
```

Comment:
I guess I have seen such code before. It seems like a independent part of logic.